lib/cmsis_rtos_v1/cmsis_timer.c

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel_structs.h>
+#include <cmsis_os.h>
+
+#define ACTIVE 1
+#define NOT_ACTIVE 0
+
+static void zephyr_timer_wrapper(struct k_timer *timer);
+
+struct timer_obj {
```

Comment:
Would the timer_obj struct be better off in a header file? I see that this is similar to what POSIX has; but since there is no `timer_obj` in CMSIS spec, is it okay to have these wrappers in user-exposed implementation files?

Reply:
well.. i don't see a need to have a separate header file for just one structure. Also, i don't see any issue with wrappers being exposed in the implementation file. We are open source anyway :)